CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter/ArrangeInScatter3dGeneralPlugin.java

```diff
@@ -0,0 +1,149 @@
+        arranger.arrange(graph);
+    }
+
+    @Override
+    public PluginParameters createParameters() {
+        final PluginParameters parameters = new PluginParameters();
+
+        final PluginParameter<SingleChoiceParameterValue> x_attribute = SingleChoiceParameterType.build(SCATTER_3D_X_ATTRIBUTE);
+        x_attribute.setName(X_ATTRIBUTE);
+        x_attribute.setDescription("The attribute to use for the x dimension");
+        x_attribute.setStringValue("");
+        parameters.addParameter(x_attribute);
+
+        final PluginParameter<SingleChoiceParameterValue> y_attribute = SingleChoiceParameterType.build(SCATTER_3D_Y_ATTRIBUTE);
+        y_attribute.setName(Y_ATTRIBUTE);
+        y_attribute.setDescription("The attribute to use for the y dimension");
+        y_attribute.setStringValue("");
+        parameters.addParameter(y_attribute);
+
+        final PluginParameter<SingleChoiceParameterValue> z_attribute = SingleChoiceParameterType.build(SCATTER_3D_Z_ATTRIBUTE);
+        z_attribute.setName(Z_ATTRIBUTE);
+        z_attribute.setDescription("The attribute to use for the z dimension");
+        z_attribute.setStringValue("");
+        parameters.addParameter(z_attribute);
+
+        final PluginParameter<BooleanParameterValue> xLogarithmic = BooleanParameterType.build(SCATTER_3D_X_LOGARITHMIC);
+        xLogarithmic.setName(X_LOGARITHMIC);
+        xLogarithmic.setDescription("Scale the X axis in Logarithmic Scale");
+        xLogarithmic.setBooleanValue(false);
+        parameters.addParameter(xLogarithmic);
+
+        final PluginParameter<BooleanParameterValue> yLogarithmic = BooleanParameterType.build(SCATTER_3D_Y_LOGARITHMIC);
+        yLogarithmic.setName(Y_LOGARITHMIC);
+        yLogarithmic.setDescription("Scale the Y axis in Logarithmic Scale");
+        yLogarithmic.setBooleanValue(false);
+        parameters.addParameter(yLogarithmic);
+
+        final PluginParameter<BooleanParameterValue> zLogarithmic = BooleanParameterType.build(SCATTER_3D_Z_LOGARITHMIC);
+        zLogarithmic.setName(Z_LOGARITHMIC);
+        zLogarithmic.setDescription("Scale the Z axis in Logarithmic Scale");
+        zLogarithmic.setBooleanValue(false);
+        parameters.addParameter(zLogarithmic);
+
+        return parameters;
+    }
+
+    @Override
+    public void updateParameters(Graph graph, PluginParameters parameters) {
+
+        // Get the list of non-default attributes
+        final ReadableGraph rg = graph.getReadableGraph();
+        Map<String, Integer> vertexAttributes = null;
+        try {
+            vertexAttributes = AttributeUtilities.getVertexAttributes(rg, 0);
+        } finally {
+            rg.release();
+        }
+
+        final List<String> keys = new ArrayList<>(vertexAttributes.keySet());
+
+        final PluginParameter<SingleChoiceParameterValue> xdimension = (PluginParameter<SingleChoiceParameterValue>) parameters.getParameters().get(SCATTER_3D_X_ATTRIBUTE);
```

Comment:
Following standard Java style, this should be `xDimension`, `yDimension` and `zDimension`.

Reply:
Done